src/js/common/schemaform/review/ReviewCollapsibleChapter.jsx

```diff
@@ -55,47 +62,60 @@ export default class ReviewCollapsibleChapter extends React.Component {
     });
   }
 
-  scrollToPage(key, index) {
-    scroller.scrollTo(`${key}${index}ScrollElement`, {
+  scrollToPage(key) {
+    scroller.scrollTo(`${key}ScrollElement`, {
       duration: 500,
       delay: 2,
       smooth: true,
     });
   }
 
   toggleChapter() {
-    const isOpening = !this.state.open;
-    this.setState({ open: !this.state.open }, () => {
-      if (isOpening) {
-        this.scrollToTop();
-      }
-    });
+    const opening = !this.state.open;
+    this.setState({ open: opening });
+
+    if (!opening) {
+      this.props.setPagesViewed(this.pageKeys);
+    }
   }
 
   render() {
     let pageContent = null;
 
-    if (this.state.open) {
-      const { form, pages } = this.props;
-      const activePages = getActivePages(pages, form.data);
-      const expandedPages = expandArrayPages(activePages, form.data);
+    const { form, pages, viewedPages, chapter } = this.props;
+    const activePages = getActivePages(pages, form.data);
+    const expandedPages = expandArrayPages(activePages, form.data);
 
+    this.pageKeys = getPageKeys(pages, form.data);
+    const hasUnViewedPages = this.pageKeys.some(key =>
```

Comment:
Should this arrow function either be brought onto one line or given curly brackets?